resdk/data_upload/annotate_samples.py

```diff
@@ -0,0 +1,85 @@
+"""Annotate samples in a collection."""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+from resdk.data_upload.samplesheet import FileImporter, Sample
+
+def annotate_samples(collection, samplesheet_name, local_sheet=True,
+                     verbose=True):
+    """Annotate the samples in a collection based on a sample annotation
+    spreadsheet.
+    """
+
+    # Extract the information from a sample annotation spreadsheet
+    # Validate each sample annotation
+    if local_sheet:
+        samplesheet = FileImporter(samplesheet_name)
+        err_samples = []
+        missing_samples = []
+        sample_count = 0
+
+    # TODO: Try to pull the samplesheet as a data object from the collection
+    #       once it is possible to upload samplesheets
+    # else:
+    #     sheet = collection.data.get(name=samplesheet_name,
+    #                                 type='data:annotation')
+    #     sheet.download()
+    #     samplesheet = FileImporter(sheet.files()[0])
+    #     err_samples = []
+
+    # Apply each annotation to a sample in the collection
+    for entry in samplesheet.sample_list:
+        try:
```

Comment:
Don't put too much code into `try`/`except` block. Only use it for lines that are expected to fail and handle specific errors.